native/common/jp_exception.cpp

```diff
@@ -210,7 +209,18 @@ void JPypeException::convertJavaToPython()
 	// Add cause to the exception
 	JPPyObject args(JPPyRef::_call, Py_BuildValue("(s)", "Java Exception"));
 	JPPyObject cause(JPPyRef::_call, PyObject_Call(PyExc_Exception, args.get(), NULL));
-	JPPyObject trace(JPPyRef::_call, PyTrace_FromJavaException(frame, th));
+	JPPyObject trace = PyTrace_FromJavaException(frame, th, NULL);
+
+	// Attach Java causes as well.
+	jthrowable jcause = frame.getCause(th);
+	if (jcause != NULL)
+	{
+		jvalue a;
+		a.l = (jobject) jcause;
+		JPPyObject prev = frame.getContext()->_java_lang_Object->convertToPythonObject(frame, a, false);
```

Comment:
can this call raise something?

Reply:
In theory yes, in practice no.   Calling convertToPythonObject can fail if the object type does not exist for some reason.  But exception types should always be wrappable unless there is something very wrong in the Python layer.   Unfortunately recovering from a double exception throw is very hard, so the best I can do here is catch the bad object  and bail from this path so that the original exception gets spat out.   

I will work on a revision.